examples/drm.rs

```diff
@@ -0,0 +1,215 @@
+                                return val == PlaneType::Primary as u32 as u64;
+                            }
+                        }
+                    }
+                }
+
+                false
+            })
+            .or(planes.first())
+            .ok_or("No planes")?;
+
+        // Create the surface on top of this plane.
+        // Note: This requires root on DRM/KMS.
+        let mut surface = unsafe {
+            Surface::from_raw(&context, {
+                let mut handle = DrmWindowHandle::empty();
+                handle.plane = (**plane).into();
+                handle.into()
+            })
+        }?;
+
+        // Resize the surface.
+        let (width, height) = mode.size();
+        surface.resize(
+            NonZeroU32::new(width as u32).unwrap(),
+            NonZeroU32::new(height as u32).unwrap(),
+        )?;
+
+        // Start drawing to it.
+        let start = Instant::now();
+        let mut tick = 0;
+        while Instant::now().duration_since(start) < Duration::from_secs(2) {
+            tick += 1;
+            println!("Drawing tick {tick}");
+
+            // Start drawing.
+            let mut buffer = surface.buffer_mut()?;
+            draw_to_buffer(&mut buffer, tick);
+            buffer.present()?;
+
+            // Wait for the page flip to happen.
+            rustix::event::poll(
+                &mut [rustix::event::PollFd::new(
+                    &device,
+                    rustix::event::PollFlags::IN,
+                )],
+                -1,
+            )?;
+
+            // Receive the events.
+            let events = device.receive_events()?;
+            println!("Got some events...");
+            for event in events {
+                match event {
+                    Event::PageFlip(_) => {
+                        println!("Page flip event.");
+                    }
+                    Event::Vblank(_) => {
+                        println!("Vblank event.");
+                    }
+                    _ => {
+                        println!("Unknown event.");
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn draw_to_buffer(buf: &mut [u32], tick: usize) {
+        let scale = colorous::SINEBOW;
+        let mut i = (tick as f64) / 20.0;
+        while i > 1.0 {
+            i -= 1.0;
+        }
+
+        let color = scale.eval_continuous(i);
+        let pixel = (color.r as u32) << 16 | (color.g as u32) << 8 | (color.b as u32);
+        buf.fill(pixel);
+    }
+
+    struct Card(std::fs::File);
+
+    impl Card {
+        fn find() -> Result<Card, Box<dyn std::error::Error>> {
+            for i in 0..10 {
+                let path = format!("/dev/dri/card{i}");
+                let device = Card::open(path)?;
+
+                // Only use it if it has connectors.
+                let handles = match device.resource_handles() {
```

Comment:
This test isn't really sufficient for a multi-gpu system since multiple GPUs will have DRM "connectors" even if they aren't physically exposed in any way.

It should check for one that's currently connected.